### ISSUE TITLE
Support excluding type-safe dependency accessors from dendencies

### ISSUE THREAD

**Issue by brainbytes42** [GradleUp/shadow#923]
### Shadow Version
8.1.1

### Gradle Version
8.6

### Expected Behavior
I'm using gradle/libs to share dependencies in my multi-project-build (see https://docs.gradle.org/current/userguide/platforms.html).
I'd like to exclude some dependencies for some sub-projects (see https://imperceptiblethoughts.com/shadow/configuration/dependencies/#filtering-dependencies)... 

### Actual Behavior
... but for example `exclude(dependency(libs.log4j.api))` raises an error: `org.gradle.internal.typeconversion.UnsupportedNotationException: Cannot convert the provided notation to an object of type Dependency: map(valueof(DependencyValueSource)).`
According to the libs-documentation (first link above), the call of `libs.log4j.api` should perform as if it was the plain string-dependency, but obviously, it doesn't.

Any ideas on this? Thanks!

**Comment by Goooler** [comment]
Any reproducer?

**Comment by Goooler** [comment]
According to your description, you want to use things like

```kt
dependencies {
  implementation(libs.log4j.api)
}

tasks.shadowJar {
  dependencies {
    exclude(libs.log4j.api)
  }
}
```

And I believe we should support project accessors as well

```kt
dependencies {
  implementation(projects.foo)
}

tasks.shadowJar {
  dependencies {
    exclude(projects.foo)
  }
}
```

See:
- https://docs.gradle.org/current/userguide/version_catalogs.html
- https://docs.gradle.org/current/userguide/declaring_dependencies_basics.html

**Comment by brainbytes42** [comment]
yeah, exactly. I have several plugins using the same dependencies as the framework, so if I could exclude some shared dependencies, the plugins would have smaller filesize.

(I actually use version catalogs as workaround, but this doesn't reduce the filesize.)

**Comment by Goooler** [comment]
You can try out the workaround

```kt
dependencies {
  implementation(libs.log4j.api)
}

tasks.shadowJar {
  dependencies {
    exclude(dependency(libs.log4j.api.get().toString()))
  }
}
```
